### DISCUSSION TITLE
Capturing When SLEAP Workflow is Completed

### DISCUSSION THREAD

**Discussion by jmdelahanty** [talmolab/sleap#973] [Help!] [answered]
Hello sleapyheads!

I was curious if there was a way to capture when SLEAP was finished running something. For example, when a prediction is finished, is there something a subprocess could capture to let you know it's done? It would be neat to have a python wrapper for spawning SLEAP jobs in parallel for a big batch of videos depending on what your computer has available on it. This is where my interest in the nvidia-smi parser thing was coming from! Currently people are writing individual shell scripts to run individual videos on different cards on a server compute node. Until we have orchestrators in place (and also since most people running SLEAP won't have orchestrators in place), it would be cool to see how we can do it on our own for a little bit.

So is there a code that SLEAP outputs if it's successful?

**Comment by roomrys** [comment, marked as answer]
Hi @jmdelahanty,

If you run the command using the [subprocess module](https://docs.python.org/3.8/library/subprocess.html), then it will return an instance of [CompletedProcess ](https://docs.python.org/3.8/library/subprocess.html#subprocess.CompletedProcess)which has an return code specifying if the command was successful or not.

**Reply by jmdelahanty** [reply]
Neat! I didn't know that was something that just sorta "worked" out of the box. I figured exit codes were something you would have to code in yourself. Thanks Liezl! Your're a legend.
